crypto/crypto.go

```diff
@@ -215,7 +230,27 @@ func NewNaiveSigner(privateKey PrivateKey, hashAlgo HashAlgorithm) NaiveSigner {
 
 // GeneratePrivateKey generates a private key with the specified signature algorithm from the given seed.
 func GeneratePrivateKey(sigAlgo SignatureAlgorithm, seed []byte) (PrivateKey, error) {
-	privKey, err := crypto.GeneratePrivateKey(crypto.SigningAlgorithm(sigAlgo), seed)
+	// check the seed has minimum entropy
+	if len(seed) < MinSeedLength {
```

Comment:
Should this check `sigAlgo.minSeedLength()`, or does it need to be the generic length?

Reply:
That needs to be the generic length. 
The specific one is done to the length of hash seed here `privKey, err := crypto.GeneratePrivateKey(crypto.SigningAlgorithm(sigAlgo), hashedSeed)` and that's guaranteed because KMAC has an arbitrary output, and we precised that we need an output length `sigAlgo.minSeedLength()`